spec.html

```diff
@@ -11151,14 +11151,14 @@ <h1>[[Construct]] ( _argumentsList_, _newTarget_ )</h1>
     </emu-clause>
 
     <emu-clause id="sec-createbuiltinfunction" aoid="CreateBuiltinFunction">
-      <h1>CreateBuiltinFunction ( _steps_, _length_, _name_, _internalSlotsList_ [ , _realm_ [ , _prototype_ [ , _prefix_ ] ] ] )</h1>
-      <p>The abstract operation CreateBuiltinFunction takes arguments _steps_, _length_, _name_, and _internalSlotsList_ (a List of names of internal slots) and optional arguments _realm_, _prototype_, and _prefix_. _internalSlotsList_ contains the names of additional internal slots that must be defined as part of the object. This operation creates a built-in function object. It performs the following steps when called:</p>
+      <h1>CreateBuiltinFunction ( _algorithm_, _length_, _name_, _internalSlotsList_ [ , _realm_ [ , _prototype_ [ , _prefix_ ] ] ] )</h1>
+      <p>The abstract operation CreateBuiltinFunction takes arguments _algorithm_, _length_ (a non-negative integer or +&infin;), _name_ (a property key), and _internalSlotsList_ (a List of names of internal slots) and optional arguments _realm_ (a Realm Record), _prototype_ (an Object or *null*), and _prefix_ (a String). _internalSlotsList_ contains the names of additional internal slots that must be defined as part of the object. This operation creates a built-in function object. It performs the following steps when called:</p>
```

Comment:
```suggestion
      <p>The abstract operation CreateBuiltinFunction takes arguments _algorithm_, _length_ (a non-negative integer or +&infin;), _name_ (a property key), _internalSlotsList_ (a List of names of internal slots), and optional arguments _realm_ (a Realm Record), _prototype_ (an Object or *null*), and _prefix_ (a String). _internalSlotsList_ contains the names of additional internal slots that must be defined as part of the object. This operation creates a built-in function object. It performs the following steps when called:</p>
```

Reply:
Actually, the precendent is to have that "and" there. See the preambles of:
- Call ( _F_, _V_ [ , _argumentsList_ ] )
- Invoke ( _V_, _P_ [ , _argumentsList_ ] )
- OrdinaryCreateFromConstructor ( _constructor_, _intrinsicDefaultProto_ [ , _internalSlotsList_ ] )
- SetFunctionName ( _F_, _name_ [ , _prefix_ ] )
- CreateBuiltinFunction ( _algorithm_, _length_, _name_, _internalSlotsList_ [ , _realm_ [ , _prototype_ [ , _prefix_ ] ] ] )
- ForIn/OfBodyEvaluation ( _lhs_, _stmt_, _iteratorRecord_, _iterationKind_, _lhsKind_, _labelSet_ [ , _iteratorKind_ ] )
- FlattenIntoArray ( _target_, _source_, _sourceLen_, _start_, _depth_ [ , _mapperFunction_, _thisArg_ ] )
- AllocateTypedArray ( _constructorName_, _newTarget_, _defaultProto_ [ , _length_ ] )
- GetValueFromBuffer ( _arrayBuffer_, _byteIndex_, _type_, _isTypedArray_, _order_ [ , _isLittleEndian_ ] )
- SetValueInBuffer ( _arrayBuffer_, _byteIndex_, _type_, _value_, _isTypedArray_, _order_ [ , _isLittleEndian_ ] )
- GetModifySetValueInBuffer ( _arrayBuffer_, _byteIndex_, _type_, _value_, _op_ [ , _isLittleEndian_ ] )
- DataView.prototype.setFoo ( _byteOffset_, _value_ [ , _littleEndian_ ] )
- PerformPromiseThen ( _promise_, _onFulfilled_, _onRejected_ [ , _resultCapability_ ] )

(or scan spec.html for " and .* and optional arg")

Reply:
Gross, ok. Will look into removing it from everywhere in a separate PR.

Reply:
I think the current form is grammatically correct. Compare:

> The state is served by representatives Alice, Bob, and Carol and senators Dan and Erin.

Here dropping the initial "and" would be clearly incorrect, because "representatives" does not describe Dan or Erin. And there's no comma after "Carol" because the following "and" is joining a list of two: "representatives and senators", not "representatives, and senators".

Anyway we should talk about it before you go to the trouble of making a new PR.

Reply:
In this case it'd be like "politicians" and "senators", because the second is a subset of the first, which is where i think the difference comes in.